### PR TITLE
feat: [FFM-9037]: Implementing the modal for mark as not stale

### PR DESCRIPTION
**Pull request by laura-mannus**
### Summary

Modal that will appear on button click of "Mark as not stale" button. Backend not hooked up yet.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/103498167/32cb972c-a6b3-4aa4-9264-c539e7a5f50e

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/pages/feature-flags/components/FeatureFlagListing.module.scss [hunk 1]
```diff
@@ -8,4 +8,11 @@
 .staleCheckbox {
   border-right: solid 0.5px var(--grey-200);
   margin-right: 15px;
+  margin-top: 10px;
+}
+
+.flagTable {
+  div[class*='TableV2--header'] {
+    align-items: center;
```
Comment (laura-mannus) [inline src/modules/75-cf/pages/feature-flags/components/FeatureFlagListing.module.scss:16]: aligning the checkbox in the header into center

File: src/modules/75-cf/pages/feature-flags/FlagStatus.tsx [hunk 2]
```diff
@@ -58,7 +62,7 @@ export const FlagStatus: React.FC<FlagStatusProps> = ({ status, lastAccess, stal
 
   return (
     <ComponentLayout spacing="xsmall" style={{ alignItems: isNeverRequested ? 'baseline' : 'center' }}>
-      {stale ? (
+      {flagCleanupEnabled && stale ? (
```
Comment (laura-mannus) [inline src/modules/75-cf/pages/feature-flags/FlagStatus.tsx:65]: forgot to wrap this in other PR

File: src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx [hunk 3]
```diff
@@ -17,6 +17,7 @@ export const StaleFlagActions: FC = () => {
   const { getString } = useStrings()
   const selectedFlags = useGetSelectedStaleFlags()
   const [showInfo, setShowInfo] = useState<boolean>(false)
+  const [notStaleOpen, setNotStaleOpen] = useState<boolean>(false)
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx:20]: this reads a bit strange for me, could it say something like `showNotStaleDialog` / `setShowNotStaleDialog` perhaps?

File: src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx [hunk 4]
```diff
@@ -30,7 +31,36 @@ export const StaleFlagActions: FC = () => {
...
+              text={getString('cf.staleFlagAction.notStale')}
+              variation={ButtonVariation.SECONDARY}
+              onClick={() => {
+                setNotStaleOpen(true)
+              }}
+            />
+            <ModalDialog
+              isOpen={notStaleOpen}
+              enforceFocus={false}
+              isCloseButtonShown={true}
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx:44]:
```suggestion
              isCloseButtonShown
```

File: src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx [hunk 5]
```diff
@@ -30,7 +31,36 @@ export const StaleFlagActions: FC = () => {
...
+              onClose={() => {
+                setNotStaleOpen(false)
+              }}
+              title={getString('cf.staleFlagAction.notStale')}
+            >
+              <Layout.Vertical>
+                <Text padding={{ bottom: 'medium' }}>{getString('cf.staleFlagAction.notStaleDesc')}</Text>
+                <Layout.Horizontal spacing="small" padding={{ top: 'large' }}>
+                  <Button text={getString('cf.staleFlagAction.notStale')} variation={ButtonVariation.PRIMARY} />
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx:53]: do we need the prop `intent={Intent.PRIMARY}` here?

File: src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx (hunk 4 above)
Comment (laura-mannus) [reply src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx]: updated

File: src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx (hunk 5 above)
Comment (laura-mannus) [reply src/modules/75-cf/pages/feature-flags/components/StaleFlagActions/StaleFlagActions.tsx]: Thanks have added

File: src/modules/75-cf/strings/strings.en.yaml [hunk 6]
```diff
@@ -291,6 +291,7 @@ staleFlagAction:
   flagLifecycleExplained: Flags lifecycle explained
   flagLifecycleDesc: Flag cleanup automation allows users to run a pipeline step that will automatically cleanup unwanted flags and any associated code within their codebase. Think Github's dependabot but for removing flags instead.
   waitingForCleanup: Waiting for cleanup
+  notStaleDesc: These Flags will be removed from the queue to cleanup and marked as not stale.
```
[the remark this answers is not in the document]
Comment (laura-mannus) [reply src/modules/75-cf/strings/strings.en.yaml:294]: I don't know I just took it exactly from the designs which I hoped Ksenia cleared with Gigi before she left

**Review by crystal-fong — APPROVED** [review]

**Review by Kira-Jennings — DISMISSED** [review]

**Review by crystal-fong — APPROVED** [review]

**Review by Kira-Jennings — APPROVED** [review]

**Pull request merged**